### PR TITLE
Feat plugins dis and en

### PR DESCRIPTION
**Pull request by struggleRoue**
### Description
https://github.com/antvis/X6/issues/3142
增加disable和enable能力

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [✅] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Enhancement (changes that improvement of current feature or performance)
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Test Case (changes that add missing tests or correct existing tests)
- [ ] Code style optimization (changes that do not affect the meaning of the code)
- [ ] Docs (changes that only update documentation)
- [ ] Chore (changes that don't modify src or test files)

### Self Check before Merge

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/antvis/x6/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/x6/src/graph/graph.ts [hunk 1]
```diff
@@ -1343,5 +1371,7 @@ export namespace Graph {
     name: string
     init: (graph: Graph, ...options: any[]) => any
     dispose: () => void
+    disable: () => void
```
Comment (NewByVector) [inline packages/x6/src/graph/graph.ts:1374]: 这里的类型可以先写为 ?: ,因为现在有很多插件没有实现 enable 和 disable 方法。

**Review by NewByVector — CHANGES_REQUESTED** [review]

File: packages/x6/src/graph/graph.ts (hunk 1 above)
Comment (struggleRoue) [reply packages/x6/src/graph/graph.ts:1376]: 可以 那我先兼容一下

File: packages/x6/src/graph/graph.ts [hunk 2]
```diff
@@ -1343,5 +1373,7 @@ export namespace Graph {
     name: string
     init: (graph: Graph, ...options: any[]) => any
     dispose: () => void
+    disable?: () => void
```
Comment (NewByVector) [inline packages/x6/src/graph/graph.ts:1376]: 这里返回值类型也需要推敲一下，例如 selection 插件中，enable 方法返回值是 this，感觉可以统一一下。

**Review by NewByVector — CHANGES_REQUESTED** [review]

File: packages/x6/src/graph/graph.ts (hunk 2 above)
Comment (struggleRoue) [reply packages/x6/src/graph/graph.ts:1376]: 你觉得统一成什么比较好？
我一般理解disable和enable不会返回this，我理解void好一点

**Review by NewByVector — CHANGES_REQUESTED** [review]

File: packages/x6/src/graph/graph.ts [hunk 3]
```diff
@@ -1198,6 +1198,53 @@ export class Graph extends Basecoat<EventArgs> {
...
+  }
+
+  disposePlugins(plugins: string[] | string) {
+    let postPlugins = plugins
+    if (!Array.isArray(postPlugins)) {
+      postPlugins = [postPlugins]
+    }
+    const aboutToChangePlugins = this.getPlugins(postPlugins)
+    aboutToChangePlugins?.forEach((plugin) => {
+      plugin?.dispose?.()
```
Comment (NewByVector) [inline packages/x6/src/graph/graph.ts:1238]: plugin?.dispose?.()   =>  plugin.dispose()

**Review by NewByVector — CHANGES_REQUESTED** [review]

**Review by NewByVector — APPROVED** [review]

**Pull request merged**